modules/sql/src/main/resources/db/migration/V038__Add_Gpi.sql

```diff
@@ -0,0 +1,386 @@
+  short_name  text        NOT NULL,
+  long_name   text        NOT NULL,
+  value       boolean     NOT NULL
+);
+
+ALTER TABLE e_gpi_pupil_camera OWNER TO postgres;
+
+--
+-- Data for Name: e_gpi_pupil_camera; Type: TABLE DATA; Schema: public; Owner: postgres
+--
+
+COPY e_gpi_pupil_camera(id, short_name, long_name, value) FROM stdin;
+In	In	In	true
+Out	Out	Out	false
+\.
+
+--
+-- Name: e_gpi_fpm; Type: TABLE; Schema: public; Owner: postgres
+--
+
+CREATE TABLE e_gpi_fpm (
+  id          identifier  PRIMARY KEY,
+  short_name  text        NOT NULL,
+  long_name   text        NOT NULL,
+  obsolete    boolean     NOT NULL
+);
+
+ALTER TABLE e_gpi_fpm OWNER TO postgres;
+
+--
+-- Data for Name: e_gpi_fpm; Type: TABLE DATA; Schema: public; Owner: postgres
+--
+
+COPY e_gpi_fpm(id, short_name, long_name, obsolete) FROM stdin;
+OPEN	Open	Open	f
+F50umPIN	50umPIN	50umPIN	f
+WITH_DOT	WITH_DOT	WITH_DOT	f
+SCIENCE	SCIENCE	SCIENCE	f
+FPM_K1	FPM_K1	FPM_K1	f
+FPM_H	FPM_H	FPM_H	f
+FPM_J	FPM_J	FPM_J	f
+FPM_Y	FPM_Y	FPM_Y	f
+\.
+
+--
+-- Name: e_gpi_sampling_mode; Type: TABLE; Schema: public; Owner: postgres
+--
+
+CREATE TABLE e_gpi_sampling_mode (
+  id          identifier PRIMARY KEY,
+  short_name  text       NOT NULL,
+  long_name   text       NOT NULL,
+  obsolete    boolean    NOT NULL
+);
+
+ALTER TABLE e_gpi_sampling_mode OWNER TO postgres;
+
+--
+-- Data for Name: e_gpi_sampling_mode; Type: TABLE DATA; Schema: public; Owner: postgres
+--
+
+COPY e_gpi_sampling_mode(id, short_name, long_name, obsolete) FROM stdin;
+FAST	Fast	Fast	f
+SINGLE_CDS	Single CDS	Single CDS	f
+MULTIPLE_CDS	Multiple CDS	Multiple CDS	f
+UTR	UTR	UTR	f
+\.
+
+--
+-- Name: e_gpi_cassegrain; Type: TABLE; Schema: public; Owner: postgres
+--
+
+CREATE TABLE e_gpi_cassegrain (
+  id          identifier PRIMARY KEY,
+  short_name  text       NOT NULL,
+  long_name   text       NOT NULL,
+  value       integer    NOT NULL
+);
+
+ALTER TABLE e_gpi_cassegrain OWNER TO postgres;
+
+--
+-- Data for Name: e_gpi_cassegrain; Type: TABLE DATA; Schema: public; Owner: postgres
+--
+
+COPY e_gpi_cassegrain(id, short_name, long_name, value) FROM stdin;
+A0	0	0	0
+A180	180	180	180
+\.
+
+--
+-- Name: e_gpi_observing_mode; Type: TABLE; Schema: public; Owner: postgres
+--
+
+CREATE TABLE e_gpi_observing_mode (
+  id                      identifier    PRIMARY KEY,
+  short_name              text          NOT NULL,
+  long_name               text          NOT NULL,
+  filter                  identifier    REFERENCES e_gpi_filter,
+  filter_iterable         boolean       NOT NULL,
+  apodizer                identifier    REFERENCES e_gpi_apodizer,
+  fpm                     identifier    REFERENCES e_gpi_fpm,
+  lyot                    identifier    REFERENCES e_gpi_lyot,
+  bright_limit_prism      numeric(4,3)  NULL,
+  bright_limit_wollaston  numeric(4,3)  NULL,
+  corresponding_h_mode    identifier    REFERENCES e_gpi_observing_mode, -- To avoid self references, null its assumed to point to itsel	f
```

Comment:
This is a bit tricky to handle. it can't be null at the end according to the old model

Reply:
I don't understand what the problem is here. A row can refer to itself.

Reply:
I think he's referring to the issue when generating code -- if he supplies as the value the same thing as the enum id then the code generation fails.  The solution instead would be to go with the `LazyRef` trick.

Reply:
Oh right, it's the generated code that's an issue. I'll pull your branch and see if I can fix it.

Reply:
Yes, it was about the generated code as I hit the compiler limitation to refer to itself. The comment is bogus as it can’t be null on the mode